presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -125,7 +125,7 @@ public PlanNodeSearcher skipOnlyWhen(Predicate<PlanNode> skipOnly)
         if (where.test(node)) {
             nodes.add((T) node);
         }
-        if (skipOnly.test(node)) {
+        if (skipOnly.test(node) || where.test(node)) {
```

Comment:
if I do not modify this `IF` then whole method `findAll` is useless as in current form it is unable to collect multiple nodes of the same type